packaging/debian/preinst

```diff
@@ -3,6 +3,7 @@
 # SPDX-License-Identifier: MIT
 
 if [ "$1" = "upgrade" ]; then
+    /opt/aws/amazon-cloudwatch-agent/bin/amazon-cloudwatch-agent-ctl -a prep-restart
```

Comment:
So, `prep-restart` is used as a mechanism to fire `cond-restart`. It's kind of weird. But `prep-restart` creates a file that is used as a flag indicator in other action, `cond-restart`, which determines whether or not to restart the agent. 
```bash
agent_cond_restart() {
    agent_name="${1:-}"
    restart_file="${2:-}"
    if [ -f "${restart_file}" ]; then
        agent_start "${agent_name}"
        rm -f "${restart_file}"
    fi
}
```
Basically, if that `${restart_file}` exists, then it will start the agent(s). And the `prep-restart` command is what creates those files.